astropy/units/format/generic.py

```diff
@@ -24,6 +24,7 @@
 from fractions import Fraction
 from typing import TYPE_CHECKING
 
+from astropy.units.errors import UnitScaleError, UnitsWarning
```

Comment:
Same question about where to import from here. And one could get rid of `core` by importing `get_current_unit_registry()`.